src/Raven.Server/Documents/Handlers/Processors/Queries/AbstractOperationQueriesHandlerProcessor.cs

```diff
@@ -31,7 +29,7 @@ protected AbstractOperationQueriesHandlerProcessor([NotNull] TRequestHandler req
     public override async ValueTask ExecuteAsync()
     {
         using (ContextPool.AllocateOperationContext(out JsonOperationContext context))
-        using (var tracker = new RequestTimeTracker(HttpContext, Logger, null, "Query"))
+        using (var tracker = new RequestTimeTracker(HttpContext, Logger, (RequestHandler as DatabaseRequestHandler)?.Database, "Query"))
```

Comment:
can we add a TODO here? looks like this requires some refactoring.

Reply:
this was handled in next commits in this PR